sale_purchase_requisition/models/purchase_requisition.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Moduon Team S.L. <info@moduon.team>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+
+
+class PurchaseRequisition(models.Model):
+    _inherit = "purchase.requisition"
+
+    sale_id = fields.Many2one(
+        comodel_name="sale.order",
+        string="Sale Order",
+        help="Sale order related to this purchase requisition.",
+    )
+    sale_user_id = fields.Many2one(
+        comoel_name="res.users",
+        string="Sale Representative",
+        related="sale_id.user_id",
+    )
+    alert_activity = fields.Boolean(default=False)
+    purchase_requisition_sync_warning = fields.Char(compute="_compute_sync_sale_order")
+
+    def _compute_sync_sale_order(self):
+        for rec in self:
+            rec.purchase_requisition_sync_warning = False
+            if rec.sale_id.purchase_requisition_sync_warning:
+                rec.purchase_requisition_sync_warning = _(
+                    "Some lines are not synced on these Sale order:"
+                    + "%s please update this lines manually" % (rec.sale_id.name),
+                )
+                if (
+                    not rec.alert_activity
+                    and len(self) == 1
+                    and (rec.user_id or rec.sale_user_id)
+                ):
+                    self.env["mail.activity"].sudo().create(
+                        {
+                            "activity_type_id": self.env.ref(
+                                "mail.mail_activity_data_todo"
+                            ).id,
+                            "user_id": rec.user_id.id or rec.sale_user_id.id,
+                            "res_id": rec.id,
+                            "res_model_id": self.env["ir.model"]
+                            .search([("model", "=", "purchase.requisition")], limit=1)
+                            .id,
+                            "summary": _("Resolve differences with the sales order"),
+                        }
+                    )
+                    rec.alert_activity = True
+            else:
+                rec.alert_activity = False
+
+
+class PurchaseRequisitionLine(models.Model):
+    _inherit = "purchase.requisition.line"
+
+    line_order_id = fields.Many2one(
+        comodel_name="sale.order.line",
+        string="Sale Order Line",
+    )
+
+    def write(self, values):
+        for rec in self:
+            if "product_qty" in values and values["product_qty"] != rec.product_qty:
+                rec.requisition_id.message_post(
+                    body=_(
+                        "The product quantity %s <br> have changed %d -> %d %s"
```

Comment:
When a translation string contains more than 1 placeholder, you must use named placeholders:

```suggestion
                        "The product quantities %(product)s <br/> have changed %(old_qty)d -> %(new_qty)d %(uom)s"
```

This is because different languages might need to position words in different orders.